x/ccv/provider/keeper/msg_server.go

```diff
@@ -309,16 +305,126 @@ func (k msgServer) SetConsumerCommissionRate(goCtx context.Context, msg *types.M
 	return &types.MsgSetConsumerCommissionRateResponse{}, nil
 }
 
-func (k msgServer) ConsumerModification(goCtx context.Context, msg *types.MsgConsumerModification) (*types.MsgConsumerModificationResponse, error) {
-	if k.GetAuthority() != msg.Authority {
-		return nil, errorsmod.Wrapf(types.ErrUnauthorized, "expected %s, got %s", k.GetAuthority(), msg.Authority)
+// CreateConsumer creates a consumer chain
+func (k msgServer) CreateConsumer(goCtx context.Context, msg *types.MsgCreateConsumer) (*types.MsgCreateConsumerResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	initGas := ctx.GasMeter().GasConsumed()
+
+	consumerId := k.FetchAndIncrementConsumerId(ctx)
+
+	k.SetConsumerOwnerAddress(ctx, consumerId, msg.Signer)
+	k.SetConsumerChainId(ctx, consumerId, msg.ChainId)
+
+	if err := k.SetConsumerMetadata(ctx, consumerId, msg.Metadata); err != nil {
+		return &types.MsgCreateConsumerResponse{}, err
+	}
+
+	// initialization parameters are optional and hence could be nil
+	if msg.InitializationParameters != nil {
+		if err := k.SetConsumerInitializationParameters(ctx, consumerId, *msg.InitializationParameters); err != nil {
+			return &types.MsgCreateConsumerResponse{}, err
+		}
+	}
+
+	// power-shaping parameters are optional and hence could be null
+	if msg.PowerShapingParameters != nil {
+		if err := k.SetConsumerPowerShapingParameters(ctx, consumerId, *msg.PowerShapingParameters); err != nil {
+			return &types.MsgCreateConsumerResponse{}, err
+		}
 	}
 
+	if k.CanLaunch(ctx, consumerId) {
+		k.SetConsumerPhase(ctx, consumerId, Initialized)
+		k.PrepareConsumerForLaunch(ctx, consumerId, nil, *msg.InitializationParameters.SpawnTime)
+	} else {
+		k.SetConsumerPhase(ctx, consumerId, Registered)
+	}
+
+	gasAfter := ctx.GasMeter().GasConsumed()
+	ctx.GasMeter().ConsumeGas(gasAfter-initGas, "creating a chain has an additional cost during spawn time, "+
```

Comment:
Why double? This seems very random.

Reply:
Agreed that it's arbitrary. I believe the discussion was that we should have something like this. Whether it should be double, triple, etc. I do not know.
My take is that, our life would be substantially simpler if `MsgCreateConsumer` was quite expensive (~200 ATOMs) in regards to thinking about what happens when we're spammed, but this ideas was not favored. Without something like this, a malicious validator could easily prevent a chain from ever launching, etc. and I do not see an easy way around this.

Reply:
Discussed offline with Marius. If this ends up a problem, we'll deal with it later on.